ArduCopter/mode_guided.cpp

```diff
@@ -162,15 +162,14 @@ void ModeGuided::vel_control_start()
     guided_mode = SubMode::Velocity;
 
     // initialise horizontal speed, acceleration
-    pos_control->set_max_speed_xy(wp_nav->get_default_speed_xy());
-    pos_control->set_max_accel_xy(wp_nav->get_wp_acceleration());
+    pos_control->set_max_speed_accel_xy(wp_nav->get_default_speed_xy(), wp_nav->get_wp_acceleration());
 
     // initialize vertical speeds and acceleration
-    pos_control->set_max_speed_z(-get_pilot_speed_dn(), g.pilot_speed_up);
-    pos_control->set_max_accel_z(g.pilot_accel_z);
+    pos_control->set_max_speed_accel_z(wp_nav->get_default_speed_down(), wp_nav->get_default_speed_up(), wp_nav->get_accel_z());
```

Comment:
Guided mode's velocity controller changes behaviour here slightly in that we now use the wpnav speed up and speed down (also note the missing negative).  I think it makes more sense to use the wpnav limits but it is a change.

Reply:
Good point but I agree that is should be based on WPNav.

set_max_speed_accel_z will accept positive or negative values for speed down. speed_down = -fabsf(speed_down);